deployments/arbitrum/usdc/migrations/1679020486_configurate_and_ens.ts

```diff
@@ -0,0 +1,321 @@
+import { BigNumber, Contract, providers } from 'ethers';
+import { expect } from 'chai';
+import { L1ToL2MessageGasEstimator } from '@arbitrum/sdk';
+import { DeploymentManager } from '../../../../plugins/deployment_manager/DeploymentManager';
+import { migration } from '../../../../plugins/deployment_manager/Migration';
+import { calldata, exp, getConfigurationStruct, proposal } from '../../../../src/deploy';
+
+const { INFURA_KEY } = process.env;
+
+const ENSName = 'compound-community-licenses.eth';
+const ENSResolverAddress = '0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41';
+const ENSSubdomainLabel = 'v3-additional-grants';
+const ENSSubdomain = `${ENSSubdomainLabel}.${ENSName}`;
+const ENSTextRecordKey = 'v3-official-markets';
+
+export default migration('1679020486_configurate_and_ens', {
+  prepare: async (_deploymentManager: DeploymentManager) => {
+    return {};
+  },
+
+  enact: async (deploymentManager: DeploymentManager, govDeploymentManager: DeploymentManager) => {
+    const trace = deploymentManager.tracer();
+    const ethers = deploymentManager.hre.ethers;
+    const { utils } = ethers;
+
+    const {
+      bridgeReceiver,
+      comet,
+      cometAdmin,
+      configurator,
+      rewards,
+    } = await deploymentManager.getContracts();
+
+    const {
+      inbox,
+      l1GatewayRouter,
+      timelock,
+      governor,
+      USDC,
+      COMP,
+    } = await govDeploymentManager.getContracts();
+
+    const l1Provider = new providers.JsonRpcProvider(`https://mainnet.infura.io/v3/${INFURA_KEY}`);
+    const l2Provider = new providers.JsonRpcProvider(`https://arbitrum-mainnet.infura.io/v3/${INFURA_KEY}`);
+    const gasEstimator = new L1ToL2MessageGasEstimator(l2Provider);
+
+    const l1GatewayRouterInterace = l1GatewayRouter.interface;
+
+    const USDCAmountToBridge = exp(10_000, 6);
+    const COMPAmountToBridge = exp(2_500, 18);
+    const usdcGatewayAddress = await l1GatewayRouter.getGateway(USDC.address);
+    const compGatewayAddress = await l1GatewayRouter.getGateway(COMP.address);
+    const refundAddress = comet.address; // XXX timelock address?
+
+    const usdcDummyCallData = l1GatewayRouterInterace.encodeFunctionData(
```

Comment:
This is the slightly unfortunate part of this migration.

For each bridged action, we have to pass in several gas-related fields (maxGas/gasLimit, gasPriceBid/maxFeePerGas, maxSubmissionCost/maxSubmissionFee, as well as the value).

One way to get those estimations is to encode the function data (with dummy values for the gas-related fields), and then pass those to function from the Arbitrum SDK (`L1ToL2MessageGasEstimator#estimateAll`). That function requires a specific type of Provider, however (a JsonRpcProvider) because it calls a few special provider functions (like `getGasPrice`). The default provider that Hardhat gives you is an `EthereumProvider`.

This solution works, but it's pretty verbose. We might be able to reuse the same dummy call data for both USDC and COMP bridging (the values are very close, and the SDK function includes some buffer).

Reply:
Hmm yeah this seems like it could be challenging to make Arbitrum proposals

Reply:
The Arbitrum docs lay out the formula for estimating gas: https://developer.arbitrum.io/devs-how-tos/how-to-estimate-gas

It might be possible to recreate the calculations from `estimateAll` on our own (and potentially without the JsonRpcProvider), but we'd still need the dummy call data

Reply:
Right we could populate with https://github.com/OffchainLabs/arbitrum-sdk/blob/main/src/lib/message/L1ToL2MessageGasEstimator.ts#L215

Reply:
reworked the gas estimation